testsuite/tests/typing-gadts/pr6158.ml

```diff
@@ -31,4 +31,6 @@ Error: This pattern matches values of type "($'a S.s, $'a S.s) eq"
        but a pattern was expected which matches values of type
          "($'a S.s, $'a S.t) eq"
        The type constructor "$'a" would escape its scope
+       Hint: "$'a" is a type variable introduced by the equation
+         "$'a S.s" = "$'a S.t"
```

Comment:
The wording and the equation might be slightly at odd in those case where `'$'a` appear on both side of the equation. Maybe a small changes would to reword it as:
>```
>        Hint: "$'a" is a type variable introduced in the equation
>         "$'a S.s" = "$'a S.t"
>```

or maybe

>```
>        Hint: "$'a" is a type variable introduced for the equation
>         "$'a S.s" = "$'a S.t"
>```

?